runtime/domain.c

```diff
@@ -1533,22 +1544,16 @@ static void caml_poll_gc_work(void)
 /* FIXME: do not raise async exceptions */
 void caml_handle_gc_interrupt(void)
 {
-  atomic_uintnat* young_limit = domain_self->interruptor.interrupt_word;
   CAMLalloc_point_here;
 
   CAML_EV_BEGIN(EV_INTERRUPT_GC);
-  if (atomic_load_acq(young_limit) == INTERRUPT_MAGIC) {
-    /* interrupt */
+  if (caml_incoming_interrupts_queued()) {
     CAML_EV_BEGIN(EV_INTERRUPT_REMOTE);
-    while (atomic_load_acq(young_limit) == INTERRUPT_MAGIC) {
-      uintnat i = INTERRUPT_MAGIC;
-      atomic_compare_exchange_strong(
-          young_limit, &i, (uintnat)Caml_state->young_start);
-    }
     caml_handle_incoming_interrupts();
     CAML_EV_END(EV_INTERRUPT_REMOTE);
   }
 
+  caml_update_young_limit();
   caml_poll_gc_work();
```

Comment:
If the conditional above only handles STW interrupts, then all other forms of interrupts should be handled by `caml_poll_gc_work`, or will not be handled by this function. (This is a bit weird, and ideally there would be some documentation of which interrupts are intended to be handled where.)

From reading the code of `caml_poll_gc_work`, my impression is that:
- synchronous minor-gc requests are  serviced, as the name suggests
- explicit GC requests from users are serviced
- "requested external interrupts" (tick thread stuff) are serviced
- but signals are *not* serviced at this point

Reply:
Please see if the added documentation on the other PR is good enough